### ISSUE TITLE
operator.itemgetter can accept any hashable type

### ISSUE THREAD

**Issue by berdario** [python/typeshed#558]
https://github.com/python/mypy/issues/2153

basically, the following is valid Python code:

    >>> itemgetter(Ellipsis)({Ellipsis: -1})
    -1

in my case I got the error message with a more mundane string key, thus yielding this error:

     No overload variant of "itemgetter" matches argument types [builtins.str]

**Comment by gvanrossum** [comment]
Oh, there are lots of interesting things we could do here. Maybe it should be a generic function, e.g.
```python
@overload
def itemgetter(item: int) -> Callable[[Sequence[T]], T]: ...
@overload
def itemgetter(item: K) -> Callable[[Mapping[K, V], V]: ...
```
except the latter doesn't work yet because it introduces a type variable in the (callable) return type (I think the reason is related to https://github.com/python/mypy/issues/1927). The `*items` case is even more interesting, see https://github.com/python/typing/issues/239.

So maybe for now we need something with a lots of `Any` (or for the arguments `object`) and a TODO comment.

**Comment by berdario** [comment]
Even if it would work, doesn't overload resolution rely only on the input arguments?

In that case

    itemgetter(1)({1: 42})

would fail to typecheck, due to the the returned function expecting a `Sequence`, not a `Mapping` (unless `Sequence[X]` are `Mapping[int, X]`)

**Comment by gvanrossum** [comment]
Fixed by #564. (Note: if you include "Fixes #558" in the body of your commit message it will auto-close the issue.)
